### DISCUSSION TITLE
Export Miden VM Verifier to Solidity

### DISCUSSION THREAD

**Discussion by jeljelih** [0xMiden/miden-vm#1180] [Q&A] [answered]
I would like to ask if it is possible to export the Miden VM Verifier to Solidity? Because I saw in Bobbin's presentation a gas cost of Solidity Verifier but I could not find information about such feature in GitHub.

**Comment by bobbinth** [comment, marked as answer]
We don't have Solidity verifier implemented yet. To support this, we'd first need to build [AirScript](https://github.com/0xPolygonMiden/air-script) backend for Solidity - so, this may take some time.

Having said that, verifying Miden VM proofs directly on chain would probably be pretty expensive (I believe the estimate I've mentioned in various places is around 3M gas per proof). A better approach is to wrap Miden VM STARK proofs in a SNARK proof - similar to how zkEVM team does it. Specifically, if a STARK proof is wrapped in FFLONK proof, I believe the on-chain verification costs goes down to something like 250K gas.

**Reply by jeljelih** [reply]
Thank you very much Bobbin for your detailed answer.
